packages/snap-preact-components/src/components/Molecules/OverlayBadge/OverlayBadge.tsx

```diff
@@ -0,0 +1,138 @@
+/** @jsx jsx */
+import { ComponentChildren, Fragment, h } from 'preact';
+
+import { jsx, css } from '@emotion/react';
+import classnames from 'classnames';
+import { observer } from 'mobx-react-lite';
+
+import { Theme, useTheme, CacheProvider } from '../../../providers';
+import { ComponentProps, StylingCSS, ComponentMap } from '../../../types';
+import { defaultBadgeComponentMap, lcm } from '../../../utilities';
+
+import type { AutocompleteController, RecommendationController, SearchController } from '@searchspring/snap-controller';
+import type { Product, ResultBadge } from '@searchspring/snap-store-mobx';
+
+const CSS = {
+	OverlayBadge: ({ meta }: OverlayBadgeProps & { meta: any }) => {
+		// TODO: type meta as MetaResponseModel when updated
+		const leftAreas = meta?.badges?.locations?.overlay?.left?.map((area: any) => area.name) || [];
+		const rightAreas = meta?.badges?.locations?.overlay?.right?.map((area: any) => area.name) || [];
+
+		const LCM = lcm(leftAreas.length, rightAreas.length);
+		const leftAreasLCM = Array.from({ length: LCM }).map((_, index) => leftAreas[Math.floor(index / (LCM / leftAreas.length))]);
+		const rightAreasLCM = Array.from({ length: LCM }).map((_, index) => rightAreas[Math.floor(index / (LCM / rightAreas.length))]);
+
+		const gridTemplateAreas = leftAreasLCM.map((left, index) => `"${left} ${rightAreasLCM[index]}"`).join(' ');
```

Comment:
I wonder if we should do this somewhere else - kinda sucks that each theme would have to repeat this logic...